### PR TITLE
pairing down http method enum a bit to more closely match http spec

### PR DESCRIPTION
**Pull request by LoganWright**

### CODE REVIEW

**Comment by tannernelson** [comment]
> Although this set can be expanded, additional methods cannot be assumed to share the same semantics for separately extended clients and servers.

Seems fair to pare this down.

**Comment by Danappelxx** [comment]
I think we should still keep `PATCH`, though. It's used commonly alongside/instead of `PUT`.

**Comment by LoganWright** [comment]
@Danappelxx good point, most servers seem to support that, as well as clients like Alamofire.

File: Sources/HTTPMethod.swift [hunk 1]
```diff
@@ -91,32 +73,8 @@ extension HTTPMethod: CustomStringConvertible {
...
-        case .Notify:          return "NOTIFY"
-        case .Subscribe:       return "SUBSCRIBE"
-        case .Unsubscribe:     return "UNSUBSCRIBE"
-        case .Patch:           return "PATCH"
-        case .Purge:           return "PURGE"
-        case .MkCalendar:      return "MKCALENDAR"
-        case .Link:            return "LINK"
-        case .Unlink:          return "UNLINK"
-        case .Raw(let method): return method.uppercaseString
+        case .Patch:           return "PATCH"ss
```
Comment (tannernelson) [inline Sources/HTTPMethod.swift]: Is the `ss` a typo?

**Comment by paulofaria** [comment]
This list is so long cause it's the sum of a lot of RFCs. And pretty popular ones. Later I'll reply with a list for each RFC and their methods. This list is inspired by nginx/nodejs http parser. So these methods are there for a good reason.

Seems to me the motivation for the pare is having a long list to switch on. This is a topic I had some thoughts about inspired by a conversation with @Danappelxx. He said it would be nice if we had a way to mark an enum as extensible. This would force a switch on this kind of enum to always contain a default clause. With this we could add cases to an enum in extensions safely. This would be a swift feature and as it doesn't exist we can't use it, but that gave me an idea. Maybe it's not even necessary but it could be a good thing.

The idea would be to add a disclaimer in HTTP method docs saying that consumers should always add a default clause when switching. This would allow us to add more cases in future versions without breaking existing code. Of course this would apply to all enums we provide. Maybe adding even more would be unnecessary for http method, but this could make sense for another enum.

Now coming back to the problem of switching a long list of cases. Could you give me an example where you would have to switch the whole list? My experience with this kind of enum is that the only places a full switch was needed is the ones we would already provide. all of the other times smaller switches with a default clause or even only `if case`s were used.

**Comment by LoganWright** [comment]
I don't have a reason explicitly to switch, just considering possible futures or implementations where someone's defining some behavior based on the method. 

I think if we link appropriate specs, this list can stay more comprehensive, but I'd prefer to keep the subset smaller. A large percentage of the ones currently included seem extremely specialized and used very rarely except in specific scenarios. I think the `Other(String)` or `Raw(String)` can encompass these edge cases well enough w/o requiring so many.

Anyone else have thoughts on this one?

**Comment by paulofaria** [comment]
Here's the list I mentioned:

https://github.com/nodejs/http-parser/blob/master/http_parser.h#L90

I don't have strong opinions about keeping all of these. But I also don't see why it would hurt.

**Pull request merged**